test/automated-suite.js

```diff
@@ -0,0 +1,212 @@
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+*/
+
+'use strict';
+
+// This is a test and we want descriptions to be useful, if this
+// breaks the max-length, it's ok.
+
+var fs = require('fs');
+var webdriver = require('selenium-webdriver');
+var chrome = require('selenium-webdriver/chrome');
+var firefox = require('selenium-webdriver/firefox');
+require('chai').should();
+
+/* eslint-disable max-len, no-console, padded-blocks, no-multiple-empty-lines */
+/* eslint-env node,mocha */
+
+// These tests make use of selenium-webdriver. You can find the relevant
+// documentation here: http://selenium.googlecode.com/git/docs/api/javascript/index.html
+
+describe('Test SW-Toolbox', () => {
+  // Driver is initialised to null to handle scenarios
+  // where the desired browser isn't installed / fails to load
+  // Null allows afterEach a safe way to skip quiting the driver
+  let driver = null;
+
+  afterEach(done => {
+    if (driver === null) {
+      return done();
+    }
+
+    driver.quit()
+    .then(() => {
+      driver = null;
+      done();
+    })
+    .thenCatch(() => {
+      driver = null;
+      done();
+    });
+  });
+
+  let checkFileExists = path => {
+    return new Promise((resolve, reject) => {
+      fs.stat(path, err => {
+        if (err) {
+          return reject(err);
+        }
+        resolve();
+      });
+    });
+  };
+
+  let performTests = (browserName, driver) => {
+    // The driver methods are wrapped in a new promise because the
+    // selenium-webdriver API seems to using some custom promise
+    // implementation that has slight behaviour differences.
+    return new Promise((resolve, reject) => {
+      driver.get('http://localhost:8888/test/')
+      .then(() => {
+        return driver.executeScript('return window.navigator.userAgent;');
+      })
+      .then(userAgent => {
+        // This is just to help with debugging so we can get the browser version
+        console.log('    Browser User Agent [' + browserName + ']: ' + userAgent);
+      })
+      .then(() => {
+        // We get webdriver to wait until window.swtoolbox.testResults is defined.
+        // This is set in the in browser mocha tests when the tests have finished
+        // successfully
+        return driver.wait(function() {
+          return driver.executeScript('return ((typeof window.swtoolbox !== \'undefined\') && window.swtoolbox.testResults !== \'undefined\');');
+        });
+      })
+      .then(() => {
+        // This simply retrieves the test results from the inbrowser mocha tests
+        return driver.executeScript('return window.swtoolbox.testResults;');
+      })
+      .then(testResults => {
+        // Resolve the outer promise to get out of the webdriver promise chain
+        resolve(testResults);
+      })
+      .thenCatch(function(err) {
```

Comment:
Nah this is webdrivers weird promises that it uses. I wrap it in a promise to contain this as much as possible.